tests/conftest.py

```diff
@@ -76,72 +91,92 @@ def caikit_nlp_runtime(grpc_server_port, http_server_port):
         },
         "log": {"formatter": "pretty"},
     }
+    if not insecure:
+        config["runtime"]["tls"] = {
+            "server": {
+                "key": SERVER_KEY_FILE,
+                "cert": SERVER_CERT_FILE,
+            }
+        }
 
     caikit.config.configure(config_dict=config)
 
 
 def get_random_port():
-    sock = socket.socket()
-    sock.bind(("", 0))
-    return sock.getsockname()[1]
+    from contextlib import closing
+
+    with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as s:
+        s.bind(("", 0))
+        s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+        port = s.getsockname()[1]
+        return port
 
 
 @pytest.fixture(scope="session")
 def grpc_server_port():
-    """port for caikit grpc runtime"""
+    """default port for caikit grpc runtime"""
     return get_random_port()
 
 
 @pytest.fixture(scope="session")
 def http_server_port():
-    """port for caikit grpc runtime"""
+    """default port for caikit grpc runtime"""
     return get_random_port()
 
 
-def channel_factory(host: str, port: int):
-    config = GrpcChannelConfig(
-        host=host,
-        port=port,
-        insecure=True,  # TODO: handle cases with secure=True with self signed certs
-    )
+def channel_factory(host: str, port: int, insecure: bool):
```

Comment:
This will probably have to be parametrized with both tls=true and mtls=true. We'll do this in a separate PR